### PR TITLE
Support collection pagination

### PR DESCRIPTION
**Pull request by nglx**
Missing bits:
- [x] add find endpoint
  - [x] panel
  - [x] panel member
  - [x] survey collection
- [x] pagination integration specs
- [x] update docs

### CODE REVIEW

File: fixtures/vcr_cassettes/panel_find.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      content-length:
+      - '145'
+      date:
+      - Wed, 13 Apr 2016 13:15:35 GMT
+      connection:
+      - close
+    body:
+      encoding: UTF-8
+      string: '{"meta":{"httpStatus":"400 - Bad Request","error":{"errorMessage":"Invalid
+        mailingListId parameter. Mailing list not found","errorCode":"GP_0"}}}'
```
Comment (nglx) [inline fixtures/vcr_cassettes/panel_find.yml:62]: That's quite weird. Qualtrics API should return 404 here to be consistent with other endpoints.

File: lib/qualtrics_api/panel_member_collection.rb [hunk 2]
```diff
@@ -22,22 +15,26 @@ def create(panel_members)
       QualtricsAPI::PanelImport.new(id: import_id, panel_id: id).propagate_connection(self)
     end
 
+    def find(id)
+      raise QualtricsAPI::NotSupported, 'Find not supported for panel member'
```
Comment (nglx) [inline lib/qualtrics_api/panel_member_collection.rb:19]: Searching for a particular panel member is not supported in Qualtrics API.

File: README.md [hunk 3]
```diff
@@ -131,14 +137,17 @@ export.status
...
 
-To get all the panels:
+To paginate all the panels:
 
 ```ruby
-QualtricsAPI.panels.fetch
+panels = QualtricsAPI.panels.fetch
+# => #<QualtricsAPI::PanelCollection:0x007f8769aae2c0 ....>
+
+panels.fetch # get next page
```
Comment (pallymore) [inline README.md:146]: this is a bit odd... let's talk about it a little bit more when i go back. 
I think the functionality of `fetch` should be consistent (always do the same thing - return the same results - and has no side effects) 
pagination should be implemented in separate methods.

Comment (nglx) [inline README.md:146]: Ok. Makes sense. And probably we should make it immutable so next/fetch is returning new object. It's also quite odd now... I'll probably divide it to sth like:
- fetch (getting initial result and storing reference to next)
- next (getting next and storing reference to next if exists)

**Comment by nglx** [comment]
Closing this PR - will open a new one to master branch.

**Pull request closed without merging**